config/cass-operator/ns-scoped/kustomization.yaml

```diff
@@ -3,7 +3,38 @@ kind: Kustomization
 
 resources:
 - github.com/k8ssandra/cass-operator/config/deployments/default?ref=master
-
+patches:
```

Comment:
Is this patch the same as the one for cluster-scoped? If so then it would be nice to have the cluster-scoped and ns-scoped kustomizations share and reuse the same patch.

Reply:
It is, but I don't know how to do what you're describing.
There seem to be 2 issues:
- patches need to be in a subdirectory of a kustomization (but we have two different directories here)
- Kustomize doesn't seem to read patches from a file, it only seems to support inline patches.

How would you do this specifically?

Reply:
What do you mean Kustomize doesn't seem to read patches from a file? Isn't that pretty common way of adding patches? At least strategicMergePatches can be in a file separately, for example in cass-operator in the default that's the case:

https://github.com/k8ssandra/cass-operator/tree/master/config/default

But even for json6902 patches that's doable, as seen here:

https://github.com/k8ssandra/cass-operator/blob/master/config/crd/kustomization.yaml#L21

Patches also don't need to be a in a subdirectory of that path, you can do a common directory. Key here is that the default parameters given to Kustomize build prevent this, but kustomize does allow overriding that. In cass-operator kustomize templates, there's plenty of these shared resources between different packages.

Run kustomize with: ``build --load-restrictor LoadRestrictionsNone `` and it should be happy to read files from pretty much anywhere.